lib/install/validate/storage.go

```diff
@@ -174,3 +184,61 @@ func (v *Validator) suggestDatastore(path string, label string, flag string) {
 		}
 	}
 }
+
+// checks if a given datastore ds is writeable by compute resource specified on cmd line --compute-resource
+func (v *Validator) targetDatastoreIsWriteable(ctx context.Context, ds *object.Datastore) bool {
+	defer trace.End(trace.Begin(ds.String()))
+
+	// get compute hosts if we haven't run this method on this object before
+	if len(v.computeHosts) == 0 {
+		var err error
+		v.computeHosts, err = v.Session.Cluster.Hosts(ctx)
+		if err != nil {
+			v.NoteIssue(err)
+			return false
+		}
+		// v.computeHosts shouldn't be zero anymore; if it still is, there's a problem
+		if len(v.computeHosts) == 0 {
+			v.NoteIssue(errors.New("Couldn't get list of hosts connected to target cluster"))
+			return false
+		}
+	}
+
+	// now get hosts connected to ds
+	attachedHosts, err := ds.AttachedClusterHosts(ctx, v.Session.Cluster)
+	if err != nil {
+		v.NoteIssue(err)
+		return false
+	}
+
+	if len(attachedHosts) == 0 {
+		// this could mean that we're not operating on a cluster..
+		log.Debugf("AttachedClusterHosts returned nothing; searching for unclustered hosts connected to datastore %s", ds.String())
+		attachedHosts, err := ds.AttachedHosts(ctx)
+
+		if err != nil {
+			v.NoteIssue(err)
+			return false
+		}
+
+		if len(attachedHosts) == 0 {
+			// shouldn't be possible to wind up here, but we should get an error message if we do ¯\_(ツ)_/¯
+			v.NoteIssue(errors.Errorf("No attached hosts found for datastore %s", ds.String()))
+			return false
+		}
+	}
```

Comment:
Same comment as above.  Lets make a helper for this and the above and hide all of it somewhere else.